### PR TITLE
docs: add README to springsdk to serve as temporary general docs

### PR DESCRIPTION
**Pull request by efgpinto**
References https://github.com/lightbend/kalix/issues/7518

### CODE REVIEW

File: sdk/spring-sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+
+Below you can find a simple example of what the `User` model for the above entity could look like:
+```java
+public class User {
+  public String email;
+  public String name;
+  
+  @JsonCreator
+  public User(@JsonProperty("email") String email, @JsonProperty("name") String name) {
```
Comment (octonato) [inline sdk/spring-sdk/README.md:53]: I know that's never clear and sometimes we find situations that it doesn't work properly. But if someone start from our archetype, they should have everything to not need to defined `@JsonProperty`. Certainly not with explicitly name.

There is one situation where `@JsonProperty` is required. That's when the constructor has only one parameter.

File: sdk/spring-sdk/README.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+The goal of the Spring SDK is to provide a code-first approach to developing Kalix applications so that a developer does not need to worry about learning protobuf upfront.
+
+This SDK offers a way to develop applications which should be familiar to someone who has worked with Spring applications by making use of annotations to assemble Kalix and its components.
+
+If you're new to Kalix and the different types of entities that exist in a Kalix app, consider reading [this section](https://docs.kalix.io/services/programming-model.html#_what_is_a_kalix_service) before continuing.  
+
+## Features
+
+As the Spring SDK is more recent than their gRPC-first counterparts, not all features are supported at this time. However, there's already a lot to play with as described next.
```
Comment (octonato) [inline sdk/spring-sdk/README.md:13]: Here we can enumerate the features that are already available.

- Action
- Value Entities
- EventSourced Entities
- Views
- Publishing and subscribing to Topics

File: sdk/spring-sdk/README.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+### Value Entities
```
Comment (octonato) [inline sdk/spring-sdk/README.md:15]: I think the first step should be to show how to start a new project using the new archetype.

**Review by octonato — COMMENTED** [review]
Looking good. Left some initial comments.

File: sdk/spring-sdk/README.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+## Upcoming features
+
+Here's a list of the features that we are working on for the short-term:
+- ACLs definition
+- Timers support
+- Unit testing
+
+And much more to come later. :)
+
+## Getting Started
```
Comment (octonato) [inline sdk/spring-sdk/README.md:222]: Yeah, better to have the Getting Started as first chapter. :-)

File: sdk/spring-sdk/README.md (hunk 2 above)
Comment (efgpinto) [reply sdk/spring-sdk/README.md]: done

File: sdk/spring-sdk/README.md (hunk 4 above)
Comment (efgpinto) [reply sdk/spring-sdk/README.md]: done

**Comment by efgpinto** [comment]
Ended-up adding some javadocs for the kalix annotations as well. Ready for review now.

File: sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Entity.java [hunk 5]
```diff
@@ -18,11 +18,24 @@
...
+ *
+ * <b>Note:</b> can only be used at type level.
+ */
 @Target(ElementType.TYPE)
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
 public @interface Entity {
+
+  /**
+   * Assign a type to the entity. This name should be unique among the different existing entities within a Kalix application.
```
[the remark this answers is not in the document]
Comment (efgpinto) [reply sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Entity.java]: indeed helpful, tks.

File: sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Query.java [hunk 6]
```diff
@@ -18,9 +18,17 @@
 
 import java.lang.annotation.*;
 
+/**
+ * Annotation used un the scope of a view for providing the query that will be used to explore data from that view.
```
[the remark this answers is not in the document]
Comment (efgpinto) [reply sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Query.java]: fixed the previous javafmt before seeing this so now had to do the change by hand. tks

File: sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Subscribe.java [hunk 7]
```diff
@@ -18,28 +18,61 @@
...
+     */
     Class<? extends kalix.javasdk.valueentity.ValueEntity<?>> value();
   }
 
+  /**
+   * Annotation for subscribing to updates from an Event-sourced Entity. It can be used only at type level.
+   * The underlying method must be declared to receive one or two parameters:
+   * <ul>
+   * <li>when one parameter is passed, the single parameter will be considered the event type such method will handle;</li>
+   * <li>when two parameters are passed, the first one will be considered the view state and the second one the event type.</li>
```
[the remark this answers is not in the document]
Comment (efgpinto) [reply sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Subscribe.java]: good call. done.

File: sdk/spring-sdk/README.md [hunk 8]
```diff
@@ -0,0 +1,510 @@
...
+
+The goal of the Spring SDK is to provide a code-first approach to developing Kalix applications so that a developer does not need to worry about learning protobuf upfront.
+
+This SDK offers a way to develop applications which should be familiar to someone who has worked with Spring applications by making use of annotations to assemble Kalix and its components.
+
+> If you're new to Kalix and the different types of entities that exist in a Kalix app, consider reading [this section](https://docs.kalix.io/services/programming-model.html#_what_is_a_kalix_service) before continuing.  
+
+## Getting Started
+
+The easiest way to get started is to use the `kalix-spring-boot-archetype` and let it generate a spring boot application for you. Try:
```
[the remark this answers is not in the document]
Comment (efgpinto) [reply sdk/spring-sdk/README.md]: tks

**Comment by efgpinto** [comment]
> Left a few comments. Ok to merge after that.

Thanks Renato. Updated now, waiting for CI to pass before merging.

**Pull request merged**